packages/amplify-graphql-auth-transformer/src/graphql-auth-transformer.ts

```diff
@@ -422,6 +424,22 @@ export class AuthTransformer extends TransformerAuthBase implements TransformerA
     });
   };
 
+  /**
+   * Amplify will not include the dynamic auth fields in the generated subscription filter input type.
+   * This is because of a limitation of AppSync service. AppSync doesn't allow multiple conditions on the same field in a filter.
+   * Amplify automatically applies filter on the dynamic auth fields, so we don't accept filters on these fields from client.
+   * @param context TransformerTransformSchemaStepContextProvider
+   */
+  removeAuthFieldsFromSubscriptionFilter = (context: TransformerTransformSchemaStepContextProvider): void => {
+    this.authModelConfig.forEach((acm, modelName) => {
+      acm.getRoles().map(role => this.roleMap.get(role)).forEach(role => {
+        if (!role.static && (role.provider === 'userPools' || role.provider === 'oidc')) {
+          removeSubscriptionFilterInputAttribute(context, modelName, role.entity);
+        }
+      });
+    });
+  };
+
```

Comment:
We remove the auth fields from the filter input type because we use these fields to apply runtime auth rules. So, we don't want customers to set custom filters on these fields.

Reply:
nit: we can add a unit test for not revealing these fields in sub filter of dynamic auth models. Besides, if I understand correctly, are these dynamic auth fields indicated as groupFields of dynamic group auth?